pentasharp/Controllers/SearchBusScheduleController.cs

```diff
@@ -0,0 +1,173 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Models.Entities;
+using pentasharp.Models.Enums;
+using pentasharp.ViewModel.BusSchedul;
+using pentasharp.ViewModel.BusReservation;
+using WebApplication1.Filters;
+using Microsoft.AspNetCore.Authorization;
+using AutoMapper.QueryableExtensions;
+using pentasharp.Models.Utilities;
+using pentasharp.Interfaces;
+using static Microsoft.EntityFrameworkCore.DbLoggerCategory.Database;
+
+namespace WebApplication1.Controllers
+{
+    [Route("api/SearchSchedule")]
+    public class SearchBusScheduleController : Controller
+    {
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IBusCompanyService _IBusCompanyService;
+        private readonly ISearchBusScheduleService _scheduleService;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public SearchBusScheduleController(AppDbContext context, IMapper mapper, IBusCompanyService busCompanyService, IHttpContextAccessor httpContextAccessor, ISearchBusScheduleService scheduleService)
+        {
+            _context = context;
+            _mapper = mapper;
+            _IBusCompanyService = busCompanyService;
+            _httpContextAccessor = httpContextAccessor;
+            _scheduleService = scheduleService;
+        }
+
+        [AllowAnonymous]
+        [HttpGet("SearchSchedules")]
+        public async Task<IActionResult> SearchSchedules(string from, string to, DateTime date)
+        {
+            var schedules = await _scheduleService.SearchSchedulesAsync(from, to, date);
+
+            if (!schedules.Any())
```

Comment:
as per other comments, move logic to the service of this endpoint and all others in this controller